File: app/src/main/java/com/example/bookapp/viewModels/ViewModelUser.java
package com.example.bookapp.viewModels;

import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

import com.example.bookapp.models.User;

public class ViewModelUser extends ViewModel {

    private MutableLiveData<User> user = new MutableLiveData<>();

    public MutableLiveData<User> getUser() {
        return user;
    }

    public void setUser(User user) {
        this.user.setValue(user);
    }
}

File: app/src/main/java/com/socialMedia/bookapp/viewModels/ViewModelUser.kt
package com.socialMedia.bookapp.viewModels

import androidx.hilt.lifecycle.ViewModelInject
import androidx.lifecycle.*
import com.socialMedia.bookapp.models.User
import com.socialMedia.dataLayer.repositories.UserRepository
import kotlinx.coroutines.launch

class ViewModelUser @ViewModelInject constructor(
        private val userRepository: UserRepository) : ViewModel() {


    val searchQuery = MutableLiveData<String>()

    val registrationMessage by lazy {
        MutableLiveData<String>()
    }


    val searchSuggestions: LiveData<List<User>> = Transformations.switchMap(searchQuery) {
        it?.let {
            return@switchMap userRepository.fetchSearchSuggestions(it)
        }
    }


}